extensions/pkg/controller/backupbucket/errorcodedetector.go

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backupbucket
+
+// ErrorCodeDetector determines the Gardener error codes for errors.
+type ErrorCodeDetector interface {
```

Comment:
What was the motivation to duplicate this interface n times for each extension kind pkg? I would rather vote to have it centrally in one place under the extension library.

Reply:
How about `extensions/pkg/controller/common` or `extensions/pkg/util`?

Reply:
IMO `extensions/pkg/controller/common/error_code_detector.go` would be fine.

Reply:
Hm, I now see that we have 1 usage of `DetermineError` in `extensions/pkg/terraformer` that is not adapted with this PR. Then IMO it would be better to place it under `extensions/pkg/util/error_code_detector.go` as it won't be used from `extensions/pkg/controller` only.